adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -1191,9 +1201,153 @@ private case class GenericGenomicRDD[T](
 
   protected def replaceRdd(
     newRdd: RDD[T],
-    newPartitionMap: Option[Array[Option[(ReferenceRegion, ReferenceRegion)]]] = None): GenericGenomicRDD[T] = {
+    isSorted: Boolean = false,
+    preservesPartitioning: Boolean = false): GenericGenomicRDD[T] = {
+
+    copy(rdd = newRdd)
+  }
+}
+
+private[rdd] object SortedGenomicRDD {
+
+  def touchFile(sc: SparkContext,
+                pathName: String) {
+
+    val path = new Path(pathName)
+    val fs = path.getFileSystem(sc.hadoopConfiguration)
+
+    val os = fs.create(path)
+    os.close()
+  }
+
+  def touchSortedMetadataFile[T, U <: GenomicRDD[T, U]](gRdd: GenomicRDD[T, U],
+                                                        sc: SparkContext,
+                                                        fileName: String) {
+
+    // is this file sorted?
+    gRdd match {
+      case _: SortedGenomicRDD[T, U] => {
+        touchFile(sc, "%s/_sorted".format(fileName))
+      }
+      case _ => { /* no-op */ }
+    }
+  }
 
-    copy(rdd = newRdd, optPartitionMap = newPartitionMap)
+  def partitionersAreCompatible(leftRdd: SortedGenomicRDD[_, _],
```

Comment:
I like the idea, but I'm not sure it's possible to make it simple enough to be worth computing. Please see my comment below regarding implementation.